From f15efc24dc1bc62b55ba1e58bf6986291cc73f34 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 30 Oct 2018 18:18:57 +0200
Subject: [PATCH 624/767] sdk_dpaa: set the skb queue mapping when looping

Save the current CPU ID on ingress, when FSL_DPAA_DBG_LOOP is set.
Use the skb_set_queue_mapping() call instead of skb_record_rx_queue()
because the stack isn't involved and won't compensate for the additional
offset.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
index 55efe41093a6..1f75f32b4fc5 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_sg.c
@@ -581,6 +581,8 @@ static inline int dpa_skb_loop(const struct dpa_priv_s *priv,
 		return 0; /* loop disabled by default */
 
 	skb_push(skb, ETH_HLEN); /* compensate for eth_type_trans */
+	/* Save the current CPU ID in order to maintain core affinity */
+	skb_set_queue_mapping(skb, raw_smp_processor_id());
 	dpa_tx(skb, dpa_loop_netdevs[priv->loop_to]);
 
 	return 1; /* Frame Tx on the selected interface */
-- 
2.17.0

